.github/workflows/ci.yml

```diff
@@ -0,0 +1,290 @@
+name: ci
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+defaults:
+  run:
+    shell: bash
+jobs:
+  check-lint:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
```

Comment:
It is a good security practice to pin the versions.

Reply:
A blogpost to provide some context for this decision 
https://michaelheap.com/ensure-github-actions-pinned-sha/

Reply:
Oh. I see. That seems kind of gross, but I get why we have to do that. I guess there's also no way to centralize those numbers so that we don't need to repeat them all over the configuration?